backend/server-api/src/main/java/org/sonarsource/sonarlint/core/serverapi/component/ComponentApi.java

```diff
@@ -82,6 +83,51 @@ private String getAllProjectsUrl() {
     return searchUrl.toString();
   }
 
+  public Optional<ProjectKeyName> getProjectKeyByProjectId(String projectId, SonarLintCancelMonitor cancelMonitor) {
+    var encodedProjectId = java.net.URLEncoder.encode(projectId, java.nio.charset.StandardCharsets.UTF_8);
+    var path = "/api/components/search_projects?projectIds=" + encodedProjectId;
+
+    try (var response = helper.rawGet(path, cancelMonitor)) {
+      if (response.isSuccessful()) {
+        var responseBody = response.bodyAsString();
+        var jsonResponse = JsonParser.parseString(responseBody).getAsJsonObject();
+        var components = jsonResponse.getAsJsonArray("components");
+
+        if (!components.isEmpty()) {
+          var firstComponent = components.get(0).getAsJsonObject();
+          var key = firstComponent.get("key").getAsString();
+          var name = firstComponent.get("name").getAsString();
+
```

Comment:
We usually read JSON in a different, more concise way. You could have a look at `SystemApi` to get inspiration